pandas/core/common.py

```diff
@@ -307,12 +307,7 @@ def is_null_slice(obj) -> bool:
     """
     We have a null slice.
     """
-    return (
-        isinstance(obj, slice)
-        and obj.start is None
-        and obj.stop is None
-        and obj.step is None
-    )
+    lib.is_null_slice(obj)
```

Comment:
Could probably directly call `is_null_slice` instead of wrapping it: `from pandas._libs.lib import is_null_slice` not sure whether that makes a noticable difference

Reply:
Certainly an option. The advantage of this is you only have one interaction with a Python object from Cython extensions, whereas the current Python code goes up to 4 runtime interactions.

Granted I don't think we call this function in a tight loop all that often, but its easier when looking at Cython annotate to not see any yellow from this call